diff -Nru multipr-2/Cargo.toml multipr-3/Cargo.toml
--- multipr-2/Cargo.toml	2024-12-14 07:28:52.235833200 -0500
+++ multipr-3/Cargo.toml	2024-12-14 07:29:48.863092300 -0500
@@ -4,7 +4,8 @@
 edition = "2021"

 [dependencies]
+patch = "0.7"

 [[bin]]
-name = "multipr"
-path = "src/main.rs"
+name = "splitpr"
+path = "src/splitpr.rs"
diff -Nru multipr-2/src/main.rs multipr-3/src/main.rs
--- multipr-2/src/main.rs	2024-12-14 07:28:52.235833200 -0500
+++ multipr-3/src/main.rs	1969-12-31 19:00:00.000000000 -0500
@@ -1,3 +0,0 @@
-fn main() {
-    println!("Hello, world!")
-}
diff -Nru multipr-2/src/splitpr.rs multipr-3/src/splitpr.rs
--- multipr-2/src/splitpr.rs	1969-12-31 19:00:00.000000000 -0500
+++ multipr-3/src/splitpr.rs	2024-12-14 07:29:48.864092200 -0500
@@ -0,0 +1,11 @@
+use std::error::Error;
+use std::io;
+
+use patch::Patch;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let input = io::read_to_string(io::stdin())?;
+    let patch = Patch::from_multiple(&input).map_err(|e| e.to_string())?;
+    println!("{:#?}", patch);
+    Ok(())
+}
